pkg/workflow/argo.go

```diff
@@ -102,77 +102,57 @@ func (a *argo) Submit(ctx context.Context, l logr.Logger, g *v1alpha1.Applicatio
 		return fmt.Errorf("applicationGroup object cannot be nil")
 	}
 
-	obj := &v1alpha12.Workflow{
-		ObjectMeta: v1.ObjectMeta{Labels: make(map[string]string)},
-	}
-
 	namespaces := []string{}
+	// Add namespaces we need to create while removing duplicates
 	for _, app := range g.Spec.Applications {
-		namespaces = append(namespaces, app.Spec.Release.TargetNamespace)
+		found := false
+		for _, namespace := range namespaces {
+			if app.Spec.Release.TargetNamespace == namespace {
+				break
+			}
+		}
+		if !found {
+			namespaces = append(namespaces, app.Spec.Release.TargetNamespace)
+		}
 	}
 
+	// Create any of the target namespaces
 	for _, namespace := range namespaces {
-		ns := corev1.Namespace{
+		ns := &corev1.Namespace{
 			ObjectMeta: v1.ObjectMeta{
 				Name: namespace,
 			},
 		}
-		// Create the namespace since helm-operator does not do this
```

Comment:
This was an issue with helm-operator. Helm Controller according this this [issue](https://github.com/fluxcd/helm-controller/issues/130) has support for the `--create-namespace` flag that helm cli supports. Outside of creating the namespace for the helm release do we have a need to manually create the namespace 🤔 ? If not then maybe all of the create ns business can be skipped since helm-controller would do it for you if set in the `HelmRelease` spec

Reply:
I agree, for now, let's keep it and remove it later. I was running into a billion issues with this codeblock because of race conditions